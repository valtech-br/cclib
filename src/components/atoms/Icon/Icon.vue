<template>
  <svg :class="['icon', `icon-${name}`]" viewBox="0 0 100 100">
    <use :xlink:href="`#icon-${name}`"></use>
  </svg>
</template>

<script>
  export default {
    name: 'icon',
    props: {
      name: {
        type: String,
        required: true
      }
    }
  }
</script>
