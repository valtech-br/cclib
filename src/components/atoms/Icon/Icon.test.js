import { mount } from '@vue/test-utils'
import Icon from './Icon.vue'

describe('Component Icon', () => {
  const Root = {
    template: `
      <div>
        <icon name="test" />
      </div>
    `
  }

  const context = {
    wrapper: null
  }

  beforeEach(() => {
    context.wrapper = mount(Root, {
      components: {
        [Icon.name]: Icon
      },
      attachTo: document.body
    })
  })

  it('Should renders correctly', () => {
    expect(context.wrapper.element).toMatchSnapshot()
  })
})
