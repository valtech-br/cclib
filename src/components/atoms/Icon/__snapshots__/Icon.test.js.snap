// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Icon Should renders correctly 1`] = `
<div>
  <svg
    class="icon icon-test"
    viewBox="0 0 100 100"
  >
    <use
      xlink:href="#icon-test"
    />
  </svg>
</div>
`;
